Primeiros passos em Python

**Imersão em Python · Pós-Graduação em Engenharia e Análise de Dados**

Nesta aula você vai:

1. executar código em um notebook;
2. usar Python como calculadora;
3. guardar valores em variáveis e reconhecer os tipos básicos;
4. montar textos com f-strings e conversar com o programa usando `input`;
5. ler mensagens de erro sem pânico;
6. construir, em grupo, uma calculadora de orçamento de viagem.

## 0. Como usar este notebook

- Antes de tudo, faça a sua cópia: **Arquivo → Salvar uma cópia no Drive**.
- Um notebook é feito de **células**. Células de texto (como esta) explicam; células de código executam.
- Para executar uma célula de código, clique nela e pressione **Shift + Enter**.
- Execute as células **de cima para baixo**. Se algo ficar estranho: **Ambiente de execução → Reiniciar sessão** e execute de novo desde o início.
- As células **Sua vez** são exercícios. Cada uma tem um nível:
  - **[Essencial]** é necessário para acompanhar a próxima aula;
  - **[Reforço]** consolida o que vimos, bom para fazer em casa;
  - **[Desafio]** é opcional, para quem quer ir além.
- Errar é parte do processo. Quando travar, use a reação vermelha na chamada ou escreva no chat.

## 1. Olá, mundo

`print` mostra algo na tela. O texto vai entre aspas.

In [1]:
print("Olá, mundo!")

Olá, mundo!


In [2]:
print("Python executa uma linha de cada vez,")
print("de cima para baixo.")

Python executa uma linha de cada vez,
de cima para baixo.


Tudo o que vem depois de `#` é um **comentário**: o Python ignora, mas pessoas leem.

In [3]:
# Esta linha é só um lembrete para humanos
print("Comentários não aparecem na saída")  # também funciona no fim da linha

Comentários não aparecem na saída


### Sua vez 1.1 [Essencial]
Mostre na tela duas linhas: o seu nome e a sua área de formação.

In [4]:
# Sua vez 1.1
# escreva seu código aqui
print('Pedro Cunha')
print('Sistemas para Internet')


Pedro Cunha
Sistemas para Internet


## 2. Python como calculadora

| Operador | Significado | Exemplo | Resultado |
| --- | --- | --- | --- |
| `+` `-` | soma e subtração | `7 + 3` | `10` |
| `*` | multiplicação | `7 * 3` | `21` |
| `/` | divisão | `7 / 2` | `3.5` |
| `//` | divisão inteira | `7 // 2` | `3` |
| `%` | resto da divisão | `7 % 2` | `1` |
| `**` | potência | `7 ** 2` | `49` |

Em Python, a casa decimal é separada por **ponto**: `3.5`, não `3,5`.

In [5]:
print(7 + 3)
print(7 / 2)
print(7 // 2)
print(7 % 2)
print(7 ** 2)

10
3.5
3
1
49


In [6]:
# A ordem das operações é a da matemática; parênteses mudam a ordem
print(2 + 3 * 4)
print((2 + 3) * 4)

14
20


No notebook, a **última linha** de uma célula é mostrada mesmo sem `print`.

In [7]:
1200 * 0.15

180.0

### Sua vez 2.1 [Essencial]
Uma assinatura custa R$ 39,90 por mês. Quanto custa em um ano? E se houver 15% de desconto no plano anual?

In [9]:
# Sua vez 2.1
# escreva seu código aqui
print(f'Assinatura custando R$ 39,90/mês em um ano custa: R$ {39.90*12:.2f} e com 15% de desconto: {(39.90*12) - ((15/100) * (39.90*12)):.2f}')

Assinatura custando R$ 39,90/mês em um ano custa: R$ 478.80 e com 15% de desconto: 406.98


### Sua vez 2.2 [Essencial]
Um filme dura 135 minutos. Use `//` e `%` para descobrir quantas horas cheias e quantos minutos restantes ele tem.

In [12]:
# Sua vez 2.2
# escreva seu código aqui
total = 135
hours = 135 // 60
mins = 135 % 60

print(f'Esse filme tem: {hours} horas e {mins} minutos')


Esse filme tem: 2 horas e 15 minutos


## 3. Variáveis: dando nome aos valores

Uma variável é uma **etiqueta** colada em um valor. O sinal `=` significa "guarde": o valor da direita vai para o nome da esquerda.

In [ ]:
preco = 39.90
meses = 12
total = preco * meses
print(total)

**Por que `478.79999999999995` e não `478.8`?** O computador guarda decimais em binário, e alguns números não cabem exatamente, assim como 1/3 não cabe em decimal. A diferença é minúscula e some quando formatamos o número para exibição, como veremos na seção 5.

In [ ]:
# Uma variável pode receber um novo valor; o antigo é esquecido
meses = 6
total = preco * meses
print(total)

**Regras para nomes**

- letras minúsculas, números e `_`; sem espaços e sem acentos: `preco_unitario`
- não pode começar com número: `2_semestre` é inválido
- maiúsculas e minúsculas são diferentes: `Total` e `total` são variáveis distintas
- escolha nomes que expliquem o conteúdo: `dias_de_viagem` é melhor do que `d`

### Sua vez 3.1 [Essencial]
Crie as variáveis `salario` (4200) e `percentual_poupanca` (0.12). Calcule quanto será poupado por mês e guarde em `poupanca_mensal`. Depois calcule `poupanca_anual` e mostre os dois valores.

In [14]:
# Sua vez 3.1
# escreva seu código aqui
salario = 4200
percentual_poupanca = 12
poupanca_mensal = (percentual_poupanca/100) * salario
poupanca_anual = poupanca_mensal * 12

print(f'Poupança mensal: R$ {poupanca_mensal:.2f}')
print(f'Poupança anual: R$ {poupanca_anual:.2f}')

Poupança mensal: R$ 504.00
Poupança anual: R$ 6048.00


## 4. Tipos de dados

| Tipo | Nome em Python | Exemplos |
| --- | --- | --- |
| número inteiro | `int` | `42`, `-3`, `0` |
| número decimal | `float` | `3.14`, `39.90` |
| texto | `str` | `"Recife"`, `'dados'` |
| verdadeiro ou falso | `bool` | `True`, `False` |

`type()` revela o tipo de qualquer valor.

In [15]:
idade = 34
altura = 1.68
cidade = "Recife"
matriculada = True

print(type(idade))
print(type(altura))
print(type(cidade))
print(type(matriculada))

<class 'int'>
<class 'float'>
<class 'str'>
<class 'bool'>


O tipo define o que cada operação faz. Observe:

In [16]:
print(2 + 2)       # soma de números
print("2" + "2")   # junção de textos
print("ha" * 3)    # repetição de texto

4
22
hahaha


### Sua vez 4.1 [Essencial]
Sem executar, anote no chat qual é o tipo de cada valor. Depois confira com `type()`.

`10` · `10.0` · `"10"` · `10 > 3`

In [1]:
# Sua vez 4.1
# escreva seu código aqui

print(type(10))  #int
print(type(10.0))#float
print(type("10"))#str
print(type(10>3))#bool

<class 'int'>
<class 'float'>
<class 'str'>
<class 'bool'>


## 5. Trabalhando com texto

In [2]:
nome = "maria clara"
print(nome.upper())
print(nome.title())
print(len(nome))                     # quantidade de caracteres
print(nome.replace("clara", "luiza"))

MARIA CLARA
Maria Clara
11
maria luiza


In [ ]:
# strip remove espaços sobrando nas pontas: muito útil ao limpar dados
resposta = "   sim  "
print(resposta.strip())

### f-strings: texto com variáveis dentro

Coloque um `f` antes das aspas e escreva as variáveis entre chaves.

In [3]:
nome = "Maria"
curso = "Engenharia e Análise de Dados"
print(f"{nome} está cursando {curso}.")

Maria está cursando Engenharia e Análise de Dados.


In [4]:
# Dentro das chaves cabe qualquer conta, e :.2f formata com 2 casas decimais
preco = 39.9
meses = 12
print(f"Total em {meses} meses: R$ {preco * meses:.2f}")

Total em 12 meses: R$ 478.80


### Sua vez 5.1 [Essencial]
Com as variáveis abaixo, mostre a frase: `Carlos, sua nota final foi 8.7 e sua situação é: aprovado.` A nota deve aparecer com uma casa decimal.

In [6]:
# Sua vez 5.1
aluno = "Carlos"
nota = 8.6666
situacao = "aprovado"
# escreva seu código aqui
print(f"{aluno}, sua nota final foi {nota:.1f} e sua situação é: {situacao}.")


Carlos, sua nota final foi 8.7 e sua situação é: aprovado.


## 6. Conversando com o programa: `input`

`input` faz uma pergunta e espera a pessoa digitar. **Atenção:** o que volta é sempre texto (`str`).

In [7]:
nome = input("Qual é o seu nome? ")
print(f"Boas-vindas, {nome}!")

Boas-vindas, label!


Para fazer contas com o que foi digitado, converta o tipo:

- `int("1990")` vira `1990`
- `float("39.9")` vira `39.9`
- `str(10)` vira `"10"`

In [8]:
ano = input("Em que ano você nasceu? ")
ano = int(ano)
idade = 2026 - ano
print(f"Em 2026 você completa {idade} anos.")

Em 2026 você completa 1825 anos.


## 7. Erros são pistas

Todo mundo que programa vê mensagens de erro todos os dias. Elas dizem **onde** e **por que** o Python parou. A técnica:

1. leia a **última linha** primeiro: ela traz o tipo do erro e a explicação;
2. procure a **seta** (`---->`), que aponta a linha com problema;
3. corrija uma coisa de cada vez e execute de novo.

Execute as células abaixo e leia cada mensagem.

In [9]:
# NameError: usar um nome que não existe (repare na letra que falta)
print(totl)

NameError: name 'totl' is not defined

In [10]:
# TypeError: misturar tipos que não combinam
idade = 34
print("Idade: " + idade)

TypeError: can only concatenate str (not "int") to str

In [11]:
# ValueError: o tipo está certo, mas o valor não serve
int("trinta")

ValueError: invalid literal for int() with base 10: 'trinta'

In [12]:
# SyntaxError: a frase está mal escrita (falta fechar o parêntese)
print("Olá"

SyntaxError: incomplete input (1375624038.py, line 2)

| Erro | O que costuma ser | Primeira coisa a conferir |
| --- | --- | --- |
| `NameError` | nome digitado errado ou célula anterior não executada | grafia e ordem de execução |
| `TypeError` | operação entre tipos incompatíveis | `type()` das variáveis |
| `ValueError` | conversão impossível, como `int("abc")` | o valor que chegou |
| `SyntaxError` | falta aspas, parêntese ou dois-pontos | a linha apontada e a anterior |
| `IndentationError` | espaços no começo da linha fora do lugar | alinhamento do bloco |

### Sua vez 7.1 [Essencial]
A célula abaixo tem **três** erros. Execute, leia a mensagem, corrija um por vez.

In [14]:
# Sua vez 7.1: conserte o programa
produto = "Caderno"
quantidade = 3
preco = "12.50"
total = quantidade * float(preco)
print(f"{quantidade} x {produto} = R$ {total:.2f}")

3 x Caderno = R$ 37.50


## 8. Prática em subgrupos: calculadora de orçamento de viagem

**Formato:** salas de até 5 pessoas · 30 minutos · uma pessoa compartilha a tela e digita, as demais orientam.

**Missão:** o programa pergunta os dados de uma viagem em grupo e mostra o orçamento.

Entradas (use `input` e converta os tipos):

1. número de dias
2. número de pessoas
3. valor da diária do hotel (o quarto é dividido por todos)
4. gasto diário com alimentação **por pessoa**
5. valor da passagem **por pessoa**

Saídas:

- custo de hospedagem, alimentação e transporte do grupo
- reserva de emergência: 10% da soma
- total geral e valor por pessoa, com duas casas decimais

Exemplo de relatório:

```
===== ORÇAMENTO DA VIAGEM =====
Hospedagem:   R$ 1100.00
Alimentação:  R$ 1800.00
Transporte:   R$ 3400.00
Reserva (10%): R$ 630.00
-------------------------------
TOTAL:        R$ 6930.00
Por pessoa:   R$ 1732.50
```

Ao voltar para a sala principal, a pessoa porta-voz conta: qual erro apareceu e como o grupo resolveu?

In [ ]:
# Prática: calculadora de orçamento de viagem
# escreva seu código aqui
dias = int(input('N de dias: '))
pessoas = int(input('N de pessoas: '))
diaria = float(input('R$ da diaria: '))
alimentacao = float(input('R$ alimentação diario por pessoa: '))
passagem = float(input('R$ da passagem por pessoa: '))

hospedagem = diaria * dias
alimentacao = alimentacao * dias * pessoas
transporte = passagem * pessoas
total = hospedagem + alimentacao + transporte
reserva = (10/100) * total

total_geral = total + reserva
por_pessoa = total_geral/pessoas
print('==== ORÇAMENTO DA VIAGEM====')
print(f'Hospedagem:    R$ {hospedagem:.2f}')
print(f'Alimentação:   R$ {alimentacao:.2f}')
print(f'Transporte:    R$ {transporte:.2f}')
print(f'Reserva (10%): R$ {reserva:.2f}')
print('----------------------------')
print(f'TOTAL:         R$ {total_geral:.2f}')
print(f'Por pessoa:    R$ {por_pessoa:.2f}')




==== ORÇAMENTO DA VIAGEM====
Hospedagem:    R$ 100.00
Alimentação:   R$ 100.00
Transporte:    R$ 100.00
Reserva (10%): R$ 30.00
----------------------------
TOTAL:         R$ 330.00
Por pessoa:    R$ 330.00


## 9. Para casa

### Reforço 9.1 [Reforço]
Converta 30 graus Celsius para Fahrenheit. Fórmula: `F = C * 9 / 5 + 32`. Mostre: `30.0 °C equivalem a 86.0 °F`.

In [15]:
# Reforço 9.1
# escreva seu código aqui
celsius = 30
fahr = celsius * 9/5 + 32
print(f"{celsius} C equivalem a {fahr}F")

30 C equivalem a 86.0F


### Reforço 9.2 [Reforço]
Uma pessoa pesa 70 kg e mede 1,75 m. Calcule o IMC (`peso / altura ** 2`) e mostre com uma casa decimal.

In [19]:
# Reforço 9.2
# escreva seu código aqui
peso = 70.0
cm = 1.75
imc = peso / (cm ** 2)
print(f"{imc:.1f}")

22.9


### Reforço 9.3 [Reforço]
Um caixa eletrônico precisa entregar R$ 390 usando o menor número de notas de 100, 50 e 20. Use `//` e `%` para descobrir quantas notas de cada valor. Confira: o resultado deve somar 390.

In [21]:
# Reforço 9.3
# escreva seu código aqui
valor_restante = 390

notas_100 = valor_restante // 100
valor_restante %= 100

notas_50 = valor_restante // 50
valor_restante %= 50

notas_20 = valor_restante // 20
valor_restante %= 20

print(notas_100*100+notas_50*50+notas_20*20)

390


### Desafio 9.4 [Desafio]
Juros compostos: quanto rendem R$ 5.000 aplicados por 3 anos a 0,8% ao mês? Fórmula: `montante = capital * (1 + taxa) ** meses`. Mostre o montante e o total de juros.

In [22]:
# Desafio 9.4
# escreva seu código aqui
capital = 5000
meses = 3 * 12
taxa = 0.8/100
montante = capital * (1+taxa) ** meses
print(f"Rendem R$ {montante:.2f}")
print(f"Total de juros R$ {montante - capital:.2f}")

Rendem R$ 6661.15
Total de juros R$ 1661.15


## Resumo da aula

- `print` mostra, `#` comenta, `=` guarda.
- Tipos básicos: `int`, `float`, `str`, `bool`. `type()` revela o tipo.
- f-strings montam textos: `f"Total: R$ {valor:.2f}"`.
- `input` sempre devolve texto; converta com `int()` ou `float()`.
- Erro não é fracasso: leia a última linha, ache a seta, corrija uma coisa por vez.

Na próxima aula: decisões (`if`), repetições (`for`) e coleções (listas e dicionários).